src/roiextractors/extractors/femtonicsimagingextractor/femtonicsimagingextractor.py

```diff
@@ -0,0 +1,343 @@
+            raise ValueError(f"MUnit {self._munit} not found in file")
+
+        attrs = dict(self._file[session_key][munit_key].attrs)
+        num_channels = attrs.get("VecChannelsSize", 2)
+
+        channels = []
+        for i in range(num_channels):
+            name_arr = attrs.get(f"Channel_{i}_Name", [])
+            name = self._decode_string(name_arr) if len(name_arr) > 0 else f"Channel_{i}"
+            channels.append(name)
+        return channels
+
+    def _decode_string(self, arr) -> str:
+        """Convert int16 array to string."""
+        if len(arr) == 0:
+            return ""
+        return "".join(chr(x) for x in arr if x != 0)
+
+    def get_image_shape(self) -> Tuple[int, int]:
+        """Get the shape of the video frame (num_rows, num_columns)."""
+        return self._video.shape[1], self._video.shape[2]  # (height, width)
+
+    def get_num_samples(self) -> int:
+        """Get the number of samples (frames) in the video."""
+        return self._video.shape[0]
+
+    def get_channel_names(self) -> List[str]:
+        """Get the channel names."""
+        return [self._selected_channel_name]
+
+    def get_sampling_frequency(self) -> float:
+        """Get the sampling frequency in Hz."""
+        time_per_frame_ms = self._get_time_per_frame()
+        if time_per_frame_ms is None:
+            raise ValueError("Sampling frequency could not be determined from metadata.")
+        return 1000.0 / time_per_frame_ms
+
+    def _get_time_per_frame(self) -> Optional[float]:
+        """Get time per frame from metadata."""
+        session_key = f"MSession_{self._munit}"
+        munit_key = f"MUnit_{self._munit}"
+        attrs = dict(self._file[session_key][munit_key].attrs)
+        return attrs.get(
+            "ZAxisConversionConversionLinearScale"
+        )  # ZAxisConversionConversionLinearScale gives the duration of one frame in milliseconds.
+
+    def get_dtype(self) -> np.dtype:
+        """Get the data type of the video."""
+        return self._video.dtype
+
+    def get_series(self, start_sample: Optional[int] = None, end_sample: Optional[int] = None) -> np.ndarray:
+        """Get a series of frames."""
+        return self._video.lazy_slice[start_sample:end_sample, :, :].dsetread()
+
+    # Femtonics-specific getter methods
+    def _get_session_uuid(self):
+        """
+        Get the session UUID from the root attributes.
+
+        Returns
+        -------
+        uuid : array or value
+            The session UUID.
+        """
+        attrs = dict(self._file.attrs)
+        return attrs.get("Uuid")
+
+    def _get_pixel_size(self) -> Tuple[float, float]:
+        """Get pixel size in micrometers."""
+        session_key = f"MSession_{self._munit}"
+        munit_key = f"MUnit_{self._munit}"
+        attrs = dict(self._file[session_key][munit_key].attrs)
+
+        x_size = attrs.get("XAxisConversionConversionLinearScale", 1.0)
```

Comment:
Actually, don't you need to decode "XAxisConversionUnitName" from the corresponding metadata to see which units this is? How does this work?

Reply:
All of this makes sense and are good decisions but I am thinking. I think it would be better to return the  x_size, y_size and the units as  strings. Just provide the metadata as it is. Then we can handle the rest in neuroconv which is where metadata like this actually matters.

 This should be handled in neuroconv where we actually are concerned about the metadata.

Reply:
And by the rest I mean the warning and, if necessary at some point, conversion.